File: widgets/selphid_plugin_v2/lib/SelphIDDocumentType.dart
class SelphIDDocumentType {

  final _value;
  const SelphIDDocumentType._internal(this._value);
  toString() => '$_value';

  static const DT_IDCARD = const SelphIDDocumentType._internal('DT_IDCard');
  static const DT_PASSPORT = const SelphIDDocumentType._internal('DT_Passport');
  static const DT_DRIVERSLICENSE = const SelphIDDocumentType._internal('DT_DriversLicense');
  static const DT_FOREIGNCARD = const SelphIDDocumentType._internal('DT_ForeignCard');
  static const DT_CREDITCARD = const SelphIDDocumentType._internal('DT_CreditCard');
  static const DT_CUSTOM = const SelphIDDocumentType._internal('DT_Custom');
}

File: widgets/selphid_plugin_v2/android/src/main/java/com/selphid/flutter/plugin/selphid_plugin/SelphidDocumentType.java
package com.selphid.flutter.plugin.selphid_plugin;

import com.facephi.fphiselphidwidgetcore.WidgetSelphIDDocumentType;

public enum SelphidDocumentType {
    DT_IDCARD("DT_IDCard") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTIDCard;
        }
    },
    DT_PASSPORT("DT_Passport") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTPassport;
        }
    },
    DT_DRIVE_LICENSE("DT_DriversLicense") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTDriversLicense;
        }
    },
    DT_FOREIGN_CARD("DT_ForeignCard") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTForeignCard;
        }
    },
    DT_CREDIT_CARD("DT_CreditCard") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTCreditCard;
        }
    },
    DT_CUSTOM("DT_Custom") {
        @Override
        public WidgetSelphIDDocumentType getDocumentType() {
            return WidgetSelphIDDocumentType.DTCustom;
        }
    };

    private final String name;

    SelphidDocumentType(String name) {
        this.name = name;
    }

    public String getName() {
        return this.name;
    }

    public abstract WidgetSelphIDDocumentType getDocumentType();
}
